pkg/karmadactl/join.go

```diff
@@ -188,16 +164,25 @@ func JoinCluster(controlPlaneRestConfig, clusterConfig *rest.Config, opts Comman
 		return err
 	}
 
-	clusterSecret, impersonatorSecret, err := obtainCredentialsFromMemberCluster(clusterKubeClient, opts.ClusterNamespace, opts.ClusterName, opts.DryRun)
+	registerOption := util.RegistryClusterOption{
+		ClusterNamespace: opts.ClusterNamespace,
+		ClusterName:      opts.ClusterName,
+		ReportSecrets:    []string{"KubeCredentials", "KubeImpersonator"},
```

Comment:
How about making `KubeCredentials`, `KubeImpersonator`, and `None` as const words.